packages/opentelemetry-types/src/trace/instrumentation/Plugin.ts

```diff
@@ -32,9 +32,60 @@ export interface Plugin<T = any> {
     moduleExports: T,
     tracer: Tracer,
     logger: Logger,
-    config?: { [key: string]: unknown }
+    config?: Partial<PluginConfig>
   ): T;
 
   /** Method to disable the instrumentation  */
   disable(): void;
 }
+
+export interface PluginConfig {
+  /**
+   * Whether to enable the plugin.
+   * @default true
+   */
+  enabled: boolean;
+
+  /**
+   * Path of the trace plugin to load.
+   * @default '@opentelemetry/plugin-http' in case of http.
+   */
+  path: string;
+
+  /**
+   * Request methods that match any string in ignoreMethods will not be traced.
+   */
+  ignoreMethods: string[];
+
+  /**
+   * URLs that partially match any regex in ignoreUrls will not be traced.
+   * In addition, URLs that are _exact matches_ of strings in ignoreUrls will
+   * also not be traced.
+   */
+  ignoreUrls: Array<string | RegExp>;
+
+  /**
+   * List of internal files that need patch and are not exported by
+   * default.
+   */
+  internalFilesExports: PluginInternalFiles;
+
+  /**
+   * If true, additional information about query parameters and
+   * results will be attached (as `attributes`) to spans representing
+   * database operations.
+   */
+  enhancedDatabaseReporting: boolean;
```

Comment:
Should this be optional?

Reply:
All the properties of `PluginConfig` are optional, see here https://github.com/open-telemetry/opentelemetry-js/blob/045e0483b8e05d47a5c494610160631dd66a8e31/packages/opentelemetry-types/src/trace/instrumentation/Plugin.ts#L43.

Reply:
What would you think about making the optionality expressed directly here in the type instead?

I find it a bit confusing that they appear to be required here and you would have to look at the `Partial<` where its used (and keep passing around `Partial<PluginConfig>` if it goes between functions)

Reply:
Ok, done in https://github.com/open-telemetry/opentelemetry-js/pull/229/commits/498be8f32427dba95f8319a888c4fb319d46343a